jmxtrans-core/src/main/java/com/googlecode/jmxtrans/scheduler/ServerScheduler.java

```diff
@@ -0,0 +1,92 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.scheduler;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.googlecode.jmxtrans.cli.JmxTransConfiguration;
+import com.googlecode.jmxtrans.executors.ExecutorRepository;
+import com.googlecode.jmxtrans.jmx.ResultProcessor;
+import com.googlecode.jmxtrans.model.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ServerScheduler {
+	private static final Logger log = LoggerFactory.getLogger(ServerScheduler.class);
+
+	private final JmxTransConfiguration configuration;
+	private final ScheduledExecutorService scheduledExecutorService;
+	private final List<ServerCommand> scheduledServerCommands = new ArrayList<>();
+	private final ExecutorRepository queryExecutorRepository;
+	private final ResultProcessor resultProcessor;
+
+	@Inject
+	public ServerScheduler(
+			@Nonnull JmxTransConfiguration configuration,
+			@Nonnull @Named("queryExecutorRepository") ExecutorRepository queryExecutorRepository,
+			@Nonnull ResultProcessor resultProcessor) {
+		this.configuration = configuration;
+		this.scheduledExecutorService = Executors.newScheduledThreadPool(configuration.getScheduledExecutorPoolSize());
+		this.queryExecutorRepository = queryExecutorRepository;
+		this.resultProcessor = resultProcessor;
+	}
+
+	public void start() {
+	}
+
+	public void schedule(Server server) {
+		ServerCommand serverCommand = new ServerCommand(server, queryExecutorRepository, resultProcessor);
+		long runPeriod = serverCommand.getRunPeriodSeconds(configuration.getRunPeriod());
+		ScheduledFuture<?> scheduledFuture = scheduledExecutorService.scheduleAtFixedRate(serverCommand, runPeriod, runPeriod, TimeUnit.SECONDS);
+		serverCommand.setScheduledFuture(scheduledFuture);
```

Comment:
Cyclic dependency bewteen serverCommand, serverScheduler and scheduledFuture slightly bothers me. Not a big deal, but it feels like we could fine a cleaner way to organize all this.